### PR TITLE
Add events to CCE

### PR DESCRIPTION
**Pull request by knelli2**
## Proposed changes

Adds two new events, `Cce::Events::ObserveFields` and `Cce::Events::ObserveTimeStep`, which are different from the GH ones because CCE is run on a singleton and has volume data written differently. Then adds these two events and the `RunEventsAndTriggers` action to the `CharacteristicEvolution` component, but only when CCM is disabled.

### Upgrade instructions

All input files for the standalone CCE executable `CharacteristicExtract` must now have an
```yaml
EventsAndTriggers:
  - Trigger:
      ...
    Events:
      ...
```
block in the input file.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/Systems/Cce/Events/ObserveFields.hpp
Comment (wthrowe) [inline src/Evolution/Systems/Cce/Events/ObserveFields.hpp]: "there are several"

File: src/Evolution/Systems/Cce/Events/ObserveFields.hpp
Comment (wthrowe) [inline src/Evolution/Systems/Cce/Events/ObserveFields.hpp]: You have an unmatched backtick at the end.  Not sure if you are missing any other intended characters as well.

File: src/Evolution/Systems/Cce/Events/ObserveFields.hpp
Comment (wthrowe) [inline src/Evolution/Systems/Cce/Events/ObserveFields.hpp:104]: Your equations aren't rendering correctly.  I don't see an obvious cause.

File: src/Evolution/Systems/Cce/Events/ObserveTimeStep.hpp
Comment (wthrowe) [inline src/Evolution/Systems/Cce/Events/ObserveTimeStep.hpp]: Does backwards CCE make sense?

**Review by wthrowe — CHANGES_REQUESTED** [review]

File: src/Evolution/Systems/Cce/Events/ObserveTimeStep.hpp
Comment (knelli2) [reply src/Evolution/Systems/Cce/Events/ObserveTimeStep.hpp]: I don't think so? Considering these are future null slices. Must have been left over when I copied the DG event. I'll get rid of it.

File: src/Evolution/Systems/Cce/Events/ObserveFields.hpp
Comment (knelli2) [reply src/Evolution/Systems/Cce/Events/ObserveFields.hpp:104]: This rendered locally for me once I fixed the missing back-tick above.

Comment (wthrowe) [reply src/Evolution/Systems/Cce/Events/ObserveFields.hpp:104]: Yep, that did it.

**Review by wthrowe — DISMISSED** [review]

**Comment by wthrowe** [comment]
Note to core devs: This is waiting on a self-requested review.

File: src/Evolution/Systems/Cce/Events/ObserveFields.hpp
Comment (Sizheng-Ma) [inline src/Evolution/Systems/Cce/Events/ObserveFields.hpp:137]: It might be nice to be more specific, say 'radial deriv'

File: src/Evolution/Systems/Cce/Events/ObserveFields.hpp
Comment (Sizheng-Ma) [inline src/Evolution/Systems/Cce/Events/ObserveFields.hpp]: It's not reasonable to consider the model data of `Tags::OneMinusY` since it's just the radial coordinate of CCE. Although practically `Tags::OneMinusY` is constant on a sphere, which subsequently means that the `(0,0)` model data is the one we want, the mode decomposition makes the data interpretation redundant. 

`Tags::OneMinusY` remains constant over time once it's created at the beginning of the simulation (collocation points). So I feel it's more reasonable to dump it once at the very beginning and then we don't have to worry about it afterwards.

**Review by Sizheng-Ma — CHANGES_REQUESTED** [review]

**Comment by Sizheng-Ma** [comment]
For future reference, the modal decomposition implemented here is on a code time slice (known as 'partially flat gauge') rather than the actual physical time (known as 'Bondi time'). This may introduce some gauge effects in the dumped data. Users should be aware of this. Maybe add this comment somewhere in the code as a warning. 

A way to get around this is to use `Cce::ScriPlusInterpolationManager`, which interpolates the data to the same Bondi time slices before the decomposition, but it will make this PR more complicated.

**Comment by knelli2** [comment]
Ok @Sizheng-Ma @wthrowe pushed a fixup so I only write `OneMinusY` once, and don't write it in modal form. I had to make a member variable `mutable` because the `operator()` of the event is const. Couldn't really think of another way.

**Review by wthrowe — CHANGES_REQUESTED** [review]
No.  No state in Events.

Write the coordinates with every observation.  That's what we do for the volume evolution, and I see no reason for this to be special.

**Comment by wthrowe** [comment]
For a little more context, mutables aren't just about thread safety.  We don't want to guarantee any particular behavior regarding recreating Events after a restart.  They may be serialized or they may be recreated from scratch, so those have to have the same result.

**Comment by knelli2** [comment]
Ok removed the state. It does seem a bit redundant to write the exact same data every time, especially when it's guaranteed to never change, but it shouldn't matter for performance so it's alright.

**Comment by Sizheng-Ma** [comment]
Looks good to me. Please squash.

**Comment by wthrowe** [comment]
Squash.

You've made writing OneMinusY optional, so if you want to only write it once you can write it using a separate trigger that only fires once at the start and leave it out of the main observation.

**Comment by knelli2** [comment]
Squashed.

**Review by Sizheng-Ma — APPROVED** [review]

**Review by wthrowe — APPROVED** [review]

**Comment by wthrowe** [comment]
CI failure is something weird about the benchmark executable, which I believe to be unrelated to this PR.

**Pull request merged**
